pkg/network/types.go

```diff
@@ -40,6 +50,25 @@ func GetNetworkInfoForEC2Instances(ec2svc ec2iface.EC2API, nodeInstanceIds []str
 		}
 	}
 
+	for asgName := range asgs {
+		asgOutput, err := autoscalingSvc.DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+			AutoScalingGroupNames: aws.StringSlice([]string{asgName}),
+		})
+		if err != nil {
+			return nil, nil, nil, nil, fmt.Errorf("Error describing autoscaling group: %s", err)
+		}
+
+		if len(asgOutput.AutoScalingGroups) == 1 {
+			asgNames = append(asgNames, asgName)
```

Comment:
Minor, for better readability i would suggest moving this Map to list conversion, down along with the other similar loops, that said we can abstract those 4 loops into something like getUniqueListFromMap()